packages/apps/api/test/routes/users_user_id_registration.test.ts

```diff
@@ -0,0 +1,104 @@
+import { mockedApi, withCallerId } from './fixtures/mockedApi';
+import { ResponseStatus } from '@weco/identity-common';
+import { randomExistingUser } from './fixtures/generators';
+
+describe('/users/{userId}/registration', () => {
+  describe('PUT /users/{userId}/registration', () => {
+    it('changes the name in Sierra', async () => {
+      const firstName = 'Jane';
+      const lastName = 'Smith';
+      const testUser = randomExistingUser({
+        firstName: 'Auth0_Registration_undefined',
+        lastName: 'Auth0_Registration_tempLastName',
+      });
+      const { api, clients } = mockedApi([testUser]);
+
+      const response = await withCallerId('@machine')(
+        api
+          .put(`/users/${testUser.userId}/registration`)
+          .send({ firstName, lastName })
+      );
+
+      expect(response.statusCode).toBe(204);
+
+      const sierraUser = await clients.sierra.getPatronRecordByRecordNumber(
+        testUser.userId
+      );
+
+      if (sierraUser.status === ResponseStatus.Success) {
+        expect(sierraUser.result.firstName).toBe(firstName);
+        expect(sierraUser.result.lastName).toBe(lastName);
+      } else {
+        throw new Error(`Unexpected failure from Sierra: ${sierraUser}`);
```

Comment:
Nit: I would rather not throw "documentary" errors like this in tests; rather I would think about the assertions/expectations that we want and then use the [expect DSL](https://jestjs.io/docs/expect) to express them. In some projects we've used https://github.com/jest-community/jest-extended to expand our vocabulary there.

Reply:
Yeah, I don't love this – it's a bit of a hack around the type system. Here `sierraUser` is `SuccessResponse | ErrorResponse`, and `result.firstName` only exists on `SuccessResponse`. 

If you go:

```typescript
expect(sierraUser.result).toBe(ResponseStatus.Success)
expect(sierraUser.result.firstName).toBe(firstName)
```

then the type checker complains that it doesn't know if `result.firstName` is defined.

I've switched it up a bit so we have type casting rather than documentary errors; which is marginally less bad.

jest-extended looks interesting but I can't find any examples where we've actually used it; might be one to revisit later.

Reply:
here's an example - it mutates the global `expect` so a bit hard to find
https://github.com/wellcomecollection/identity/blob/main/packages/shared/sierra-client/tests/http-sierra-client.test.ts#L141